Highlight support

AlexzundeR opened the pull request:
I expand your core, add new criteria, add new fields to base entities that needs to support highlight request and response. Add new type of materializer, that can preprocess response after it become collection of entities (to do this I had to change logic of process query by main visitor, that can now recognize that here is chainMaterializer and create simple response processing chain).
PS I am team lead on project that is very interested in your library, and I want to help you to improve it and make it more usefull and beatyfull =)
PPS sory for my English =)

AlexzundeR committed:
Added support to create requests with highlight attribute

AlexzundeR committed:
Forgot return some deletions

damieng commented:
Hi @AlexzundeR welcome to the project!

This feature looks interesting and we'd love to take this further. I have a few points:
1. Unit tests - I didn't see any unit tests in this pull request. We have comprehensive test coverage and want to keep coverage high so would need tests for this new feature.
2. There is no need for HighlightCriteria to implement ICriteria (that's for query and filter predicates) - probably worth merging HighlightConfig into the class rather than having a separate one after that.

I'll check out the code a bit more this weekend - let me know if you need any clarification on these points.

Thanks!

Commit:
Remove highlight criteria, replace it with highlighConfig in code.

Commit:
Add some tests to Highlight extention method. Insert some assertion logic in highlight extentions. Create posibility to made chain of calls Highlight extention method.

Commit:
Add test to sure that config must be set on last call of Highlight

AlexzundeR commented:
Hi @damieng  !
1. I added some tests, I hope they will help to understand how working logic of highlighting.
2. I understood you. There was realy no need in HighlightCriteria, and now there is no HighlightCriteria class in project =) All usings of HighlightCriteria changed to HighlightConfig.

I added functionality of chain calling of my extention, to add many properties to highlighting to request.

Thank for feedback=)
I'll wait for your checkout!

AlexzundeR committed:
Add tests for highlight materializer

AlexzundeR committed:
Merge branch 'highlightSupport' of https://github.com/AlexzundeR/ElasticLINQ into highlightSupport

AlexzundeR commented:
I added some tests for Materializers. 
Now I think that my Highlight materializer is not materializer in fact, it is more like decorator for response, which changes it before all materializers start process response.
